persistence/src/test/java/cz/muni/fi/pa165/mushrooms/dao/MushroomDaoTest.java

```diff
@@ -0,0 +1,260 @@
+package cz.muni.fi.pa165.mushrooms.dao;
+
+
+import cz.muni.fi.pa165.mushrooms.entity.Mushroom;
+import cz.muni.fi.pa165.mushrooms.enums.MushroomType;
+import cz.muni.fi.pa165.mushrooms.validation.PersistenceSampleApplicationContext;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.junit4.AbstractJUnit4SpringContextTests;
+import org.springframework.test.context.transaction.TransactionalTestExecutionListener;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.PersistenceContext;
+import javax.persistence.PersistenceException;
+import javax.persistence.PersistenceUnit;
+import javax.validation.ConstraintViolationException;
+
+import java.util.Date;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.*;
+
+/**
+ * @author bencikpeter
+ */
+@ContextConfiguration(classes= PersistenceSampleApplicationContext.class)
+@TestExecutionListeners(TransactionalTestExecutionListener.class)
+@Transactional
+public class MushroomDaoTest extends AbstractJUnit4SpringContextTests {
+
+    @Autowired
+    private MushroomDao mushroomDao;
+
+    @PersistenceContext
+    private EntityManager em;
+
+
+
+    private Mushroom mushroom1;
+    private Mushroom mushroom2;
+
+    @Before
+    public void setUp(){
+        // create mushrooms
+        mushroom1 = createMushroom("toadstool", MushroomType.POISONOUS,"June","July");
+        mushroom2 = createMushroom("champignon", MushroomType.EDIBLE,"May","September");
+
+        // persist mushrooms
+
+        em.persist(mushroom1);
+        em.persist(mushroom2);
+    }
+
+    private static Mushroom createMushroom(String name, MushroomType type, String beginMonth,String endMonth){
+        Mushroom mushroom = new Mushroom();
+        mushroom.setName(name);
+        mushroom.setType(type);
+        mushroom.setIntervalOfOccurrence(beginMonth,endMonth);
+
+        return mushroom;
+    }
+
+    @Test
+    public void findById_validId() throws Exception {
+        Mushroom mushroom = mushroomDao.findById(mushroom1.getId());
+        assertThat(mushroom).isNotNull();
+        assertThat(mushroom).isEqualToComparingFieldByField(mushroom1);
+    }
+
+    @Test
+    public void findById_invalidId() throws Exception {
+        Mushroom mushroom = mushroomDao.findById(mushroom1.getId()+20);
```

Comment:
relies on a particular ID generating strategy in the db (what if db generated id for mushroom2 was mushroom1.getId()+20?). Possible solutions:
* add all known ID's
* work on an empty database